src/main/java/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionService.java

```diff
@@ -72,7 +82,21 @@ public TranslatedNonMatchingResponseBody translateSuccessResponse(List<Assertion
 
         validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance, levelOfAssurance);
 
+        validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance, levelOfAssurance);
+
+        AuthnStatement authnStatement = authnAssertion.getAuthnStatements().get(0);
         String nameID = mdsAssertion.getSubject().getNameID().getValue();
+        String issuerId =  authnAssertion.getIssuer().getValue();
+
+        String hashId = userIdHashFactory.hashId(
```

Comment:
At least we are calculating the hashId now, but we are still not using it!  The problem is further down in the method, but Github won't let me put a comment there.  We are still using the raw nameId when we build the response body; instead, we should be using the hashId you have generated here.

We are also going to need to add a test which checks that we actually use the hashId; at the moment we only have one which checks that we generate it, which is why this hasn't been spotted.

Reply:
You were right nameId now replaced by the calculated hashId, this also lead to failing acceptance tests; the ones that assert on pid so I removed these assert statements. As they are now covered by a new explicit acceptance test `responseShouldGenerateHashValueForSpecifiedPid`